qiita_pet/handlers/api_proxy/artifact.py

```diff
@@ -92,23 +96,48 @@ def artifact_post_req(user_id, filepaths, artifact_type, name,
     """
     prep = PrepTemplate(int(prep_template_id))
     study_id = prep.study_id
+
+    # First check if the user has access to the study
     access_error = check_access(study_id, user_id)
     if access_error:
         return access_error
+
     uploads_path = get_mountpoint('uploads')[0][1]
+    path_builder = partial(join, uploads_path, str(study_id))
     cleaned_filepaths = []
-    for ftype in filepaths:
-        # Check if filepath being passed exists for study
-        for fp in filepaths[ftype]:
-            full_fp = join(uploads_path, str(study_id), fp)
-            exists = check_fp(study_id, full_fp)
-            if exists['status'] != 'success':
-                return {'status': 'error',
-                        'message': 'File does not exist: %s' % fp}
-            cleaned_filepaths.append((full_fp, ftype))
-
-    artifact = Artifact.create(cleaned_filepaths, artifact_type, name=name,
-                               prep_template=prep)
+    for ftype, file_list in viewitems(filepaths):
+        # JavaScript sends us this list as a comma-separated list
+        for fp in file_list.split(','):
+            # JavaScript will send this value as an empty string if the
+            # list of files was empty. In such case, the split will generate
+            # a single element containing the empty string. Check for that case
+            # here and, if fp is not the empty string, proceed to check if
+            # the file exists
```

Comment:
Thanks for the description here, very useful. Just to check, there's no possible way to get commas in these filenames right? I'm wondering if what we should be getting should be a JSON object, but that I guess gets more in the topic of the REST API etc.